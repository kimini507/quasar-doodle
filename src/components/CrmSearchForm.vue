<template>
  <q-input
    :value="search"

    :placeholder="placeholder"
    @input="emitSearch"
    input-class="search-form__input"
    :debounce="300"
  >
    <template #append>
      <q-btn
        @click="emitSearch(search)"
        class="search-form__submit-button"
      />
    </template>
  </q-input>
</template>

<script>
import { QInput, QBtn } from 'quasar'

export default {
  components: {
    QInput,
    QBtn
  },

  props: {
    placeholder: {
      type: String,
      default: 'Search'
    },

    search: {
      type: String,
      default: ''
    }
  },

  methods: {
    emitSearch (val) {
      this.$emit('search', val)
    }
  }
}
</script>

<style>

</style>
