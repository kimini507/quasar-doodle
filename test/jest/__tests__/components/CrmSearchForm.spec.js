/* eslint-disable */
/**
 * @jest-environment jsdom
 */

import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import CrmSearchForm from 'src/components/CrmSearchForm'
import { Quasar, QInput, debounce } from 'quasar'
import flushPromises from 'flush-promises'

let wrapper = null

const searchString = 'my search criteria'
const localVue = createLocalVue()

localVue.use(Quasar, {
  components: {
    QInput
  }
})

beforeEach(() => {
  wrapper = mount(CrmSearchForm, {
    localVue,
    propsData: {
      search: searchString,
      placeholder: 'Search Area'
    }
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('CrmSearchForm', () => {
  it('renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders a search input', () => {
    expect(wrapper.find('input.search-form__input').exists()).toBeTruthy()
  })

  it('sets the placeholder of the search input', () => {
    expect(wrapper.find('input.search-form__input').attributes('placeholder')).toBe('Search Area')
  })

  it('emits search', () => {
    wrapper.vm.emitSearch(searchString)

    expect(wrapper.emitted().search).toBeTruthy()
    expect(wrapper.emitted().search[0]).toEqual([searchString])
  })

  it('emits search on update of value', () => {
    wrapper.find('input.search-form__input').setValue('updated value')

    debounce(() => {
      expect(wrapper.emitted().search).toBeTruthy()
      expect(wrapper.emitted().search[0]).toEqual(['updated value'])
    }, 300)
  })

  it('emits search on click of button', () => {
    wrapper.find('button.search-form__submit-button').trigger('click')

    expect(wrapper.emitted().search).toBeTruthy()
    expect(wrapper.emitted().search[0]).toEqual([searchString])
  })
})
