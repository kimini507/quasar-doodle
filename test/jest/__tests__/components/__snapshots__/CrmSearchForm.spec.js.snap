// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrmSearchForm renders the component 1`] = `
<div class="q-field row no-wrap items-start q-input q-field--standard q-field--float">
  <div class="q-field__inner relative-position col self-stretch column justify-center">
    <div tabindex="-1" class="q-field__control relative-position row no-wrap">
      <div class="q-field__control-container col relative-position row no-wrap q-anchor--skip"><input tabindex="0" placeholder="Search Area" type="text" class="q-field__native q-placeholder search-form__input"></div>
      <div class="q-field__append q-field__marginal row no-wrap items-center"><button tabindex="0" type="button" class="q-btn inline q-btn-item non-selectable search-form__submit-button q-btn--standard q-btn--rectangle q-focusable q-hoverable">
          <div tabindex="-1" class="q-focus-helper"></div>
          <div class="q-btn__content text-center col items-center q-anchor--skip justify-center row"></div>
        </button></div>
    </div>
  </div>
</div>
`;
